### PR TITLE
[ACS-8425] Add functionality to select folder and assign it to a hold

### PR DESCRIPTION
**Pull request by DaryaBalvanovich**
**Please check if the PR fulfills these requirements**

https://hyland.atlassian.net/browse/ACS-8425

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
Add behavior for folder

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: docs/content-services/services/legal-hold.service.md [hunk 1]
```diff
@@ -33,19 +35,35 @@ Manages holds for nodes.
...
+
+-   **bulkHold**(holdId: `string`, query: `string`, language: `string`): [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`void`]`>`<br/>
+    Assign a node to a hold.
+    -   _holdId_: `string` - The hold Id
+    -   _query_: `string` - Search query
+    -   _language_: `string` - Language code. `afts` can be used to automatically define a language
+
+    -   **Returns** [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`BulkHoldAddResponse`](../../../lib/js-api/src/api/gs-core-rest-api/docs/BulkHoldAddResponse.md)`>` - Bulk status <br/>
+
+-   **bulkFolderHold**(holdId: `string`, query: `string`, language: `string`): [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`void`]`>`<br/>
```
Comment (MichalKinas) [inline docs/content-services/services/legal-hold.service.md:60]: What is the difference between this method and the one above? From this perspective they look exactly the same

File: lib/content-services/src/lib/legal-hold/services/legal-hold.service.ts [hunk 2]
```diff
@@ -117,4 +117,16 @@ export class LegalHoldService {
     bulkHold(holdId: string, query: string, language: string): Observable<BulkHoldAddResponse> {
```
Comment (MichalKinas) [inline lib/content-services/src/lib/legal-hold/services/legal-hold.service.ts:117]: Can we keep the naming consistent and use `bulkAssignHold` or `bulkAssignHolds`?

File: lib/content-services/src/lib/legal-hold/services/legal-hold.service.ts [hunk 3]
```diff
@@ -117,4 +117,16 @@ export class LegalHoldService {
...
+
+    /**
+     * Assign a folder to a hold.
+     *
+     * @param holdId The identifier of a hold
+     * @param query Search query
+     * @param language Language Code
+     * @returns Observable<BulkHoldAddResponse>
+     */
+    bulkFolderHold(holdId: string, query: string, language: string): Observable<BulkHoldAddResponse> {
```
Comment (MichalKinas) [inline lib/content-services/src/lib/legal-hold/services/legal-hold.service.ts:129]: This method looks basically the same as the one above, why do we need a separate one?

File: docs/content-services/services/legal-hold.service.md (hunk 1 above)
Comment (DaryaBalvanovich) [reply docs/content-services/services/legal-hold.service.md]: add specific query

File: lib/content-services/src/lib/legal-hold/services/legal-hold.service.ts (hunk 2 above)
Comment (DaryaBalvanovich) [reply lib/content-services/src/lib/legal-hold/services/legal-hold.service.ts]: renamed to `bulkAssignHold`

File: docs/content-services/services/legal-hold.service.md [hunk 4]
```diff
@@ -33,19 +35,35 @@ Manages holds for nodes.
...
+
+-   **bulkAssignHold**(holdId: `string`, query: `string`, language: `string`): [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`void`]`>`<br/>
+    Assign a node to a hold.
+    -   _holdId_: `string` - The hold id
+    -   _query_: `string` - Search query
+    -   _language_: `string` - Language code. `afts` can be used to automatically define a language
+
+    -   **Returns** [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`BulkAssignHoldResponse`](../../../lib/js-api/src/api/gs-core-rest-api/docs/BulkAssignHoldResponse.md)`>` - Bulk status <br/>
+
+-   **bulkAssignHoldToFolder**(holdId: `string`, query: `string`, language: `string`): [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`void`]`>`<br/>
```
[the remark this answers is not in the document]
Comment (DaryaBalvanovich) [reply docs/content-services/services/legal-hold.service.md]: done, thanks for catching!

File: docs/content-services/services/legal-hold.service.md [hunk 5]
```diff
@@ -33,19 +35,35 @@ Manages holds for nodes.
...
+    Assign a node to a hold.
```
[the remark this answers is not in the document]
Comment (DaryaBalvanovich) [reply docs/content-services/services/legal-hold.service.md]: updated

**Review by MichalKinas — APPROVED** [review]

**Review by g-jaskowski — APPROVED** [review]

**Review by g-jaskowski — APPROVED** [review]

File: lib/content-services/src/lib/legal-hold/services/legal-holds.service.spec.ts [hunk 6]
```diff
@@ -153,18 +153,36 @@ describe('LegalHoldsService', () => {
...
+                done();
+            });
+        });
+    });
+
+    describe('bulkAssignHoldToFolder', () => {
+        it('should add nodes to hold based on search query results', (done) => {
+            const nodeId = 'mockNodeId';
+            const folderId = 'mockFolderId';
+            const language = SEARCH_LANGUAGE.AFTS;
```
[the remark this answers is not in the document]
Comment (DaryaBalvanovich) [reply lib/content-services/src/lib/legal-hold/services/legal-holds.service.spec.ts]: refactored

File: docs/content-services/services/legal-hold.service.md [hunk 7]
```diff
@@ -33,19 +35,34 @@ Manages holds for nodes.
...
     Assign a node to a hold.
     -   _holdId_: `string` - The hold Id
     -   _nodeId_: `string` - The Id of the node which is unassigned
-    -   **Returns** [`void`]
+    -   **Returns** [`void`] <br/>
+
+-   **bulkAssignHold**(holdId: `string`, query: `string`, language: `string`): [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`void`]`>`<br/>
+    Assign multiple files to a hold.
+    -   _holdId_: `string` - The hold id
+    -   _query_: `<`[`RequestQuery`](../../../lib/js-api/src/api/search-rest-api/docs/RequestQuery.md)`>` - Array of holds that should be created
```
[the remark this answers is not in the document]
Comment (DaryaBalvanovich) [reply docs/content-services/services/legal-hold.service.md]: thanks for catching! updated

**Review by swapnil-verma-gl — APPROVED** [review]

File: docs/content-services/services/legal-hold.service.md [hunk 8]
```diff
@@ -33,19 +35,33 @@ Manages holds for nodes.
...
+    -   **Returns** [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`HoldPaging`](../../../lib/js-api/src/api/gs-core-rest-api/docs/Hold.md)`>` - Hold paging <br/>
 
 -   **unassignHold**(holdId: `string`, nodeId: `string`): [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`void`]`>`<br/>
     Assign a node to a hold.
     -   _holdId_: `string` - The hold Id
     -   _nodeId_: `string` - The Id of the node which is unassigned
-    -   **Returns** [`void`]
+    -   **Returns** [`void`] <br/>
+
+-   **bulkAssignHold**(holdId: `string`, query: [`RequestQuery`](../../../lib/js-api/src/api/search-rest-api/docs/RequestQuery.md)): [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`void`]`>`<br/>
```
Comment (MichalKinas) [inline docs/content-services/services/legal-hold.service.md:52]: Return type doesn't match

File: docs/content-services/services/legal-hold.service.md [hunk 9]
```diff
@@ -33,19 +35,33 @@ Manages holds for nodes.
...
+    Assign multiple files to a hold.
+    -   _holdId_: `string` - The hold id
+    -   _query_: [`RequestQuery`](../../../lib/js-api/src/api/search-rest-api/docs/RequestQuery.md) - Search query
+    -   **Returns** [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`BulkAssignHoldResponse`](../../../lib/js-api/src/api/gs-core-rest-api/docs/BulkAssignHoldResponse.md)`>` - Bulk status <br/>
+
+-   **bulkAssignHoldToFolder**(holdId: `string`, folderId: `string`, language: `string`): [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`void`]`>`<br/>
```
Comment (MichalKinas) [inline docs/content-services/services/legal-hold.service.md:58]: Return type doesn't match

Comment (DaryaBalvanovich) [reply docs/content-services/services/legal-hold.service.md]: fixed

File: docs/content-services/services/legal-hold.service.md (hunk 8 above)
Comment (DaryaBalvanovich) [reply docs/content-services/services/legal-hold.service.md]: fixed

**Pull request merged**